src/ia-dropdown.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  html,
+  css,
+  LitElement,
+  TemplateResult,
+  nothing,
+  svg,
+  SVGTemplateResult,
+} from 'lit';
+import { property, customElement } from 'lit/decorators.js';
+
+export interface optionInterface {
+  url?: string;
+  selectedHandler?: Function;
+  label: string | TemplateResult;
+  id: string;
+}
+
+@customElement('ia-dropdown')
+export class IaDropdown extends LitElement {
+  @property({ type: Boolean, attribute: true }) open = false;
+
+  @property({ type: Boolean, attribute: true }) displayCaret = false;
+
+  @property({ type: String, attribute: true }) selectedOption = '';
+
+  @property({ type: Array }) options = [];
+
+  @property({ type: String }) optionGroup: string = 'options';
+
+  @property({ type: Function }) optionSelected = () => {};
+
+  renderOption(availableOption: optionInterface): TemplateResult {
+    const { label, url = undefined, id } = availableOption;
+    let component;
+    const selected = this.selectedOption === id ? 'selected' : '';
+
+    if (url) {
+      component = html`<a
+        href=${url}
+        @click=${() => this.optionClicked(availableOption)}
+        >${label}</a
+      >`;
+    } else {
+      component = html`<button
+        @click=${() => this.optionClicked(availableOption)}
+      >
+        ${label}
+      </button>`;
+    }
+
+    return html`<li class=${selected}>${component}</li>`;
+  }
+
+  optionClicked(option: optionInterface): void {
+    this.selectedOption = option.id;
+
+    this.dispatchEvent(
+      new CustomEvent('optionSelected', {
+        detail: { option },
+      })
+    );
+
+    if (option.selectedHandler) {
+      option?.selectedHandler(option);
+    }
+  }
+
+  toggleOptions() {
+    if (!this.open) {
+      this.open = true;
+      return;
+    }
+
+    this.open = false;
```

Comment:
You can make this bunch code in single liner fix.
`this.open = !this.open;`

Reply:
thanks!